scripts/lib/aws/elasticache.sh

```diff
@@ -21,6 +21,68 @@ print_k8s_ack_controller_pod_logs() {
   kubectl logs -n ack-system "$ack_ctrl_pod_id"
 }
 
+# To be called upon test failure: prints resource state from k8s,
+#   aws resource state, and k8s controller logs. The correct aws CLI describe call is
+#   identified from the resource plural in k8s_resource_name.
+
+# log_and_exit requires 1 or 2 arguments:
+#   k8s_resource_name: the full name of the k8s resource, including the plural (e.g. "replicationgroups/rg-1")
+#   aws_name_override (optional): used in the AWS CLI describe call if the .metadata.name field does not match
+#                                 the AWS resource name
+log_and_exit() {
+  if [[ $# < 1 || $# >2 ]]; then
+    echo "FATAL: Wrong number of arguments passed to ${FUNCNAME[0]}"
+    echo "Usage: ${FUNCNAME[0]} k8s_resource_name [aws_name_override]"
+    exit 1
+  fi
+  local k8s_resource_name=$1
+
+  # validate and process input
+  if [[ "$k8s_resource_name" != *"/"* ]]; then
+    echo "FATAL: k8s_resource_name must be of form 'plural/resource_name'"
+    exit 1
+  fi
+  local resource_plural=$(echo $k8s_resource_name | cut -f 1 -d "/")
+  local aws_resource_name=$(echo $k8s_resource_name | cut -f 2 -d "/")
+  aws_resource_name="${2:-$aws_resource_name}" # apply aws_name_override if provided
+
+  # controller logs
+  echo
+  print_k8s_ack_controller_pod_logs
+
+  # k8s resource state
+  printf "\nk8s object state:\n"
+  kubectl get "$k8s_resource_name" -o json
+
+  printf "\nAWS CLI describe call\n"
+  # aws CLI describe call
+  case $resource_plural in
```

Comment:
As the conditions do lowercase matching consider doing:
```
resource_plural=$(shell echo $(resource_plural) | tr '[:upper:]' '[:lower:]')
```
before the case statement.